From 8f237dde7fe7ad6f61fee08900f7aa2e72f3af47 Mon Sep 17 00:00:00 2001
From: Andrii Kurdiumov <kant2002@gmail.com>
Date: Tue, 29 Aug 2023 17:29:13 +0600
Subject: [PATCH] Save work

---
 src/buildtools/buildtools.targets             |  2 +-
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/buildtools/buildtools.targets b/src/buildtools/buildtools.targets
index f6a2664fe..cd61a3996 100644
--- a/buildtools/buildtools.targets
+++ b/buildtools/buildtools.targets
@@ -21,7 +21,7 @@ type Construct =
           BeforeTargets="CoreCompile">
 
     <PropertyGroup>
-        <FsLexPath Condition="'$(FsLexPath)' == ''">$(ArtifactsDir)\bin\fslex\Release\net10.0\linux-x64\fslex.dll</FsLexPath>
+        <FsLexPath Condition="'$(FsLexPath)' == ''">$(ArtifactsDir)\bin\fslex\Release\net10.0\ubuntu.22.04-x64\fslex.dll</FsLexPath>
     </PropertyGroup>
 
     <!-- Create the output directory -->
@@ -44,7 +44,7 @@ type Construct =
           BeforeTargets="CoreCompile">
 
     <PropertyGroup>
-        <FsYaccPath Condition="'$(FsYaccPath)' == ''">$(ArtifactsDir)\bin\fsyacc\Release\net10.0\linux-x64\fsyacc.dll</FsYaccPath>
+        <FsYaccPath Condition="'$(FsYaccPath)' == ''">$(ArtifactsDir)\bin\fsyacc\Release\net10.0\ubuntu.22.04-x64\fsyacc.dll</FsYaccPath>
     </PropertyGroup>
 
     <!-- Create the output directory -->
-- 
2.37.1.windows.1

